src/plugins/kibana_usage_collection/server/collectors/usage_counters/integration_tests/rollups.test.ts

```diff
@@ -0,0 +1,205 @@
+      adjustTimeout: (t: number) => jest.setTimeout(t),
+    });
+
+    esServer = await startES();
+    root = createRootWithCorePlugins();
+
+    await root.preboot();
+    await root.setup();
+    const start = await root.start();
+
+    logger = root.logger.get('test daily rollups');
+    internalRepository = start.savedObjects.createInternalRepository([
+      USAGE_COUNTERS_SAVED_OBJECT_TYPE,
+    ]);
+  });
+
+  it('deletes documents older that the retention period, from all namespaces', async () => {
+    // insert a bunch of usage counters in multiple namespaces
+    const old = [
+      createCounter('domain1', 'a', USAGE_COUNTERS_KEEP_DOCS_FOR_DAYS + 1),
+      createCounter('domain1', 'b', USAGE_COUNTERS_KEEP_DOCS_FOR_DAYS + 1, 'one'),
+      createCounter('domain1', 'b', USAGE_COUNTERS_KEEP_DOCS_FOR_DAYS + 1, 'two'),
+      createCounter('domain2', 'a', USAGE_COUNTERS_KEEP_DOCS_FOR_DAYS + 1),
+    ];
+
+    const recent = [
+      createCounter('domain1', 'a', USAGE_COUNTERS_KEEP_DOCS_FOR_DAYS - 1),
+      createCounter('domain1', 'b', USAGE_COUNTERS_KEEP_DOCS_FOR_DAYS - 1, 'one'),
+      createCounter('domain1', 'b', USAGE_COUNTERS_KEEP_DOCS_FOR_DAYS - 1, 'two'),
+      createCounter('domain2', 'a', USAGE_COUNTERS_KEEP_DOCS_FOR_DAYS - 1),
+      createCounter('domain2', 'c', USAGE_COUNTERS_KEEP_DOCS_FOR_DAYS - 1), // different counterName (does not have an old counterpart)
+    ];
+
+    getCurrentTimeMock.mockReturnValue('2024-06-24T10:00:00.000Z'); // 6 days old
+    await Promise.all(old.map((counter) => incrementCounter(internalRepository, counter)));
+
+    getCurrentTimeMock.mockReturnValue('2024-06-26T10:00:00.000Z'); // 4 days old
+    await Promise.all(recent.map((counter) => incrementCounter(internalRepository, counter)));
+
+    // check that all documents are there
+    const beforeRollup = await internalRepository.find<UsageCountersSavedObjectAttributes>({
+      type: USAGE_COUNTERS_SAVED_OBJECT_TYPE,
+      namespaces: ['*'],
+    });
+    expect(
+      beforeRollup.saved_objects
+        .map(({ attributes, updated_at: updatedAt, namespaces }) =>
+          serializeCounterKey({ ...attributes, date: updatedAt, namespace: namespaces?.[0] })
+        )
+        .sort()
+    ).toEqual(ALL_COUNTERS);
+
+    // run the rollup logic
+    isSavedObjectOlderThanMock.mockImplementation(
+      ({ doc }) => doc.updated_at === '2024-06-24T10:00:00.000Z'
+    );
+    await rollUsageCountersIndices(logger, internalRepository);
+
+    // check only recent counters are present
+    const afterRollup = await internalRepository.find<UsageCountersSavedObjectAttributes>({
+      type: USAGE_COUNTERS_SAVED_OBJECT_TYPE,
+      namespaces: ['*'],
+    });
+    expect(
+      afterRollup.saved_objects
+        .map(({ attributes, updated_at: updatedAt, namespaces }) =>
+          serializeCounterKey({ ...attributes, date: updatedAt, namespace: namespaces?.[0] })
+        )
+        .sort()
+    ).toEqual(RECENT_COUNTERS);
```

Comment:
Added an integration test to ensure we delete old ones, across namespaces, but preserve recent ones with keys matching those of the old ones.